File: src/com/fictional_particle_sim/util/VectorSpace.java
package com.fictional_particle_sim.util;

import javax.swing.*;
import java.awt.*;

public class VectorSpace extends JFrame {

    private final TheCanvas canvas = new TheCanvas();


    public VectorSpace() {

        setLayout(new BorderLayout());
        setSize(Constants.WIDTH, Constants.HEIGHT);
        setTitle("Vector Space");
        add("Center", canvas);
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

        setLocationRelativeTo(null);
        setVisible(true);
    }
    public VectorSpace(int width, int height) {

        setLayout(new BorderLayout());
        setSize(width, height);
        setTitle("Vector Space");
        add("Center", canvas);
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

        setLocationRelativeTo(null);
        setVisible(true);
    }
}

File: src/com/fictional_particle_sim/util/VectorSpace.kt
package com.fictional_particle_sim.util

import javax.swing.JFrame
import java.awt.BorderLayout

class VectorSpace : JFrame {

    private val canvas = TheCanvas()

    constructor() {
        layout = BorderLayout()
        setSize(Constants.WIDTH, Constants.HEIGHT)
        title = "Vector Space"
        add("Center", canvas)
        defaultCloseOperation = EXIT_ON_CLOSE
        setLocationRelativeTo(null)
        isVisible = true
    }
    constructor(width: Int, height: Int) {
        layout = BorderLayout()
        setSize(width, height)
        title = "Vector Space"
        add("Center", canvas)
        defaultCloseOperation = EXIT_ON_CLOSE
        setLocationRelativeTo(null)
        isVisible = true
    }
}